main.py

```
@@ -1134,0 +1,1230 @@
+    def __init__(self, cache_dir: str = ".request_cache", max_age: int = 3600):
+        """
+        Initialize the request cache.
+
+        Args:
+            cache_dir: Directory to store cached responses
+            max_age: Maximum age of cached responses in seconds (default: 1 hour)
+        """
+        self.cache_dir = Path(cache_dir)
+        self.cache_dir.mkdir(parents=True, exist_ok=True)
+        self.max_age = max_age
+        self.memory_cache: Dict[
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)